python/cudf/cudf/core/window/rolling.py

```diff
@@ -205,6 +208,33 @@ def _apply_agg_series(self, sr, agg_name):
                 self.center,
                 agg_name,
             )
+        elif isinstance(self.window, BaseIndexer):
+            start, end = self.window.get_window_bounds(
+                num_values=len(self.obj),
+                min_periods=self.min_periods,
+                center=self.center,
+                closed=None,
+            )
+            start = as_column(start, dtype="int32")
+            end = as_column(end, dtype="int32")
```

Comment:
Note that this does a HtoD copy of a potentially large array. Thus my comment https://github.com/rapidsai/cudf/pull/9085/files#r695093462